src/ngraph/runtime/interpreter/int_backend.cpp

```diff
@@ -15,97 +15,325 @@
 *******************************************************************************/
 
 #include "ngraph/runtime/interpreter/int_backend.hpp"
-#include "ngraph/runtime/host_tensor_view.hpp"
-#include "ngraph/runtime/interpreter/int_call_frame.hpp"
-#include "ngraph/runtime/interpreter/int_external_function.hpp"
+#include "ngraph/descriptor/layout/dense_tensor_view_layout.hpp"
+#include "ngraph/op/convert.hpp"
+#include "ngraph/op/select.hpp"
+#include "ngraph/op/util/binary_elementwise_comparison.hpp"
+#include "ngraph/pass/assign_layout.hpp"
+#include "ngraph/pass/liveness.hpp"
+#include "ngraph/pass/manager.hpp"
+#include "ngraph/util.hpp"
 
-using namespace ngraph;
 using namespace std;
+using namespace ngraph;
+
+using descriptor::layout::DenseTensorViewLayout;
 
 static bool static_init()
 {
     runtime::Backend::register_backend("INTERPRETER",
-                                       make_shared<runtime::interpreter::INT_Backend>());
+                                       make_shared<runtime::interpreter::INTBackend>());
     return true;
 };
 
-bool runtime::interpreter::INT_Backend::init = static_init();
-
-shared_ptr<runtime::interpreter::INT_CallFrame> runtime::interpreter::INT_Backend::make_call_frame(
-    const shared_ptr<runtime::interpreter::ExternalFunction>& external_function)
-{
-    return external_function->make_call_frame();
-}
+bool runtime::interpreter::INTBackend::init = static_init();
 
 shared_ptr<runtime::TensorView>
-    runtime::interpreter::INT_Backend::create_tensor(const element::Type& element_type,
-                                                     const Shape& shape)
+    runtime::interpreter::INTBackend::create_tensor(const element::Type& type, const Shape& shape)
 {
-    return make_shared<runtime::HostTensorView>(element_type, shape, "external");
+    return make_shared<runtime::HostTensorView>(type, shape, "external");
 }
 
-shared_ptr<runtime::TensorView> runtime::interpreter::INT_Backend::create_tensor(
-    const element::Type& element_type, const Shape& shape, void* memory_pointer)
+shared_ptr<runtime::TensorView> runtime::interpreter::INTBackend::create_tensor(
+    const element::Type& type, const Shape& shape, void* memory_pointer)
 {
-    return make_shared<runtime::HostTensorView>(element_type, shape, memory_pointer, "external");
+    return make_shared<runtime::HostTensorView>(type, shape, memory_pointer, "external");
 }
 
-bool runtime::interpreter::INT_Backend::compile(shared_ptr<Function> func)
+bool runtime::interpreter::INTBackend::compile(shared_ptr<Function> function)
 {
-    FunctionInstance& instance = m_function_map[func];
-    if (instance.m_external_function == nullptr)
+    FunctionInstance& instance = m_function_map[function];
+    if (!instance.m_is_compiled)
```

Comment:
I like the old check for nullptr.  let's just kill the is_compiled flag and check for nullptr here.